via/static/scripts/video_player/hooks/use-side-by-side-layout.ts

```diff
@@ -0,0 +1,93 @@
+import { useCallback, useEffect, useState } from 'preact/hooks';
+
+// `hypothesis:layoutchange` is triggered on the host document `body` element by
+// the client's `annotator`. Types are copied from the `client` project. See
+// https://github.com/hypothesis/client/blob/16e37788fdc52fd7e7b8eb9c0eece6a993334424/src/annotator/events.ts
+type SidebarLayout = {
+  /** Whether sidebar is open or closed */
+  expanded: boolean;
+  /** Current width of sidebar in pixels */
+  width: number;
+  /** Current height of sidebar in pixels */
+  height: number;
+  /** Width of controls (toolbar, bucket bar) on the edge of the sidebar */
+  toolbarWidth: number;
+};
+
+type LayoutChangeEventDetail = {
+  sideBySideActive: boolean;
+  sidebarLayout: SidebarLayout;
+};
+
+type LayoutChangeEvent = CustomEvent & { detail: LayoutChangeEventDetail };
+
+// Minimum body width at which laying out side-by-side should be considered
+export const SIDEBYSIDE_THRESHOLD = 875;
+// The sidebar reports its toolbar width as 33px. This is a custom padding
+// amount that allows this app to nestle up more cloesly to the sidebar when
+// closed.
+export const TOOLBAR_WIDTH = 22;
+
+/**
+ * Adjust the available space to the right of app content to make room for
+ * the sidebar when plausible. This mimics the client's "side-by-side" mode,
+ * tuned specifically for this application.
+ *
+ * Side-by-side layout will be re-computed whenever sidebar state via the
+ * `hypothesis:layoutchange` event changes meaningfully, or when the window
+ * is resized.
+ *
+ * @param [containerEl] test seam
```

Comment:
```suggestion
 * @param [container] test seam
```

Parameter name is out of date